### DISCUSSION TITLE
AR Marker : Feature

### DISCUSSION THREAD

**Discussion by burungiu** [google/model-viewer#3645] [Ideas]
Hello, guys

There is a way to use marker to anchor a GLB/GLTF model in AR mode? I checked documentation and there is not specified anywhere about this feature. It's scheduled for future development or there is a way to use it at this moment?

Thank you!

**Comment by elalish** [comment]
Can you be more specific? I'm not sure what feature exactly you want. Perhaps a description of your use case and some images?

**Comment by burungiu** [comment]
https://www.youtube.com/watch?v=ciXC3hxadBg&ab_channel=TravisFeirtag
I'm asking about something like in video above

**Comment by elalish** [comment]
Ah thank you, much clearer. I believe we're calling this the image tracking API and there's some discussion of it in the webXR khronos group. Once a spec is settled and browsers start to implement it, I'd be interested in integrating it here. If you have thoughts on the best way to do that in our declarative API, feel free to chime in here.

**Comment by mithru** [comment]
@burungiu while we can't do that on Android yet, if you want to try it on iOS you can do it by using `.reality` files (which you can use in modelviewer's `ios-src` instead of `.usdz`). 

Create the `.reality` from a `usdz` file using Apple's Reality composer, set the image target here as well. then export the `.reality` file

Unfortunately, it's ios only, but thought you would find it interesting.

**Comment by roy-baker** [comment]
@mithru @burungiu Once you get the correct Mime-Type set 

> model/vnd.reality

a .reality file with an anchor will work with ModelViewer code as mentioned. But... there is one small issue: if the marker goes out of view of the iOS camera then the model will revert to using the ground plane and drift until the marker comes back into view.

**Comment by phaseOne** [comment]
See also this discussion: https://github.com/google/model-viewer/discussions/2612

WebXR spec is called [the Image Tracking Module (aka Marker Tracking)](https://github.com/immersive-web/marker-tracking/blob/main/explainer.md). See also the somewhat related [Anchors Module](https://github.com/immersive-web/anchors/blob/main/explainer.md). Neither of these specs seem to have much implementation behind them at this time.

**Comment by burungiu** [comment]
Hello, do we have updates on this topic? I'm searching for solution on Android side, for iOS everything works perfectly in Safari with ".reality" object.

**Reply by anakin2022** [reply]
just silence for android... :(

**Comment by Fatima-Aslam** [comment]
Can someone provide me with the complete code to augment the model on the target image for both IOS and android?
